### DISCUSSION TITLE
Yellow screen when accessing webserver

### DISCUSSION THREAD

**Discussion by erikpt** [PlummersSoftwareLLC/NightDriverStrip#167] [Q&A]
Just setup a fresh environment and got everything built and uploaded.  When I access the webserver, I only see a yellow screen with "React Router Example" in the tab title. I'm using the heltec_wifi_kit_32_v2 board with 8MB flash. 

It looks like the spiffs.bin file will not build using the default size of 0x130000 so the only change I made to the partitions_custom.csv was to increase the storage partition from 0x130000 to 0x180000.  This allowed me to build and upload the storage fs but the webpage never loads.

Any pointers would be appreciated.

What I see when accessing the webserver:
![image](https://user-images.githubusercontent.com/4423348/204971986-c47789e0-018e-48da-9dc6-6e885583ad42.png)

**Comment by erikpt** [comment]
For further context, I also tried the partition boundaries used here by @burginh
https://github.com/PlummersSoftwareLLC/NightDriverStrip/discussions/159#discussioncomment-4247605

**Comment by burginh** [comment]
what browser are you using?  I had issues with Safari, but Chrome worked.  Well enough to load the web page.  but the web page doesn't look right for me (latin text) and no function.

**Reply by erikpt** [reply]
Tried Chrome current, tried Edge current, tried Firefox current.  When you access the webpage you can see the available memory (heltec so it has a screen) go from 60k to 52k, it downloads the index.html, and the main.css file, but the main.js file never downloads, the browser just hangs waiting for data.  Going to look at the packets w/Wireshark later to see if it's getting anything at all from main.js.

**Reply by erikpt** [reply]
Looks like the client just keeps waiting for nightdriver to send main.js
![image](https://user-images.githubusercontent.com/4423348/205084674-ef55d3f8-7c7a-4329-99f3-516dcfb54393.png)

**Comment by burginh** [comment]
Are you sure the file is on the spiffs ?

I added this to function from a spiffs example.

void listAllFiles(){
  Serial.println("List All Files");
  File root = SPIFFS.open("/");
 
  File file = root.openNextFile();
 
  while(file){
      Serial.print("FILE: ");
      Serial.println(file.name());
      file = root.openNextFile();
  }
 
}


Then added this to the main loop (at the bottom)

EVERY_N_SECONDS(10) {
         listAllFiles();
}

that will prove the files is there to be served.

**Comment by davepl** [comment]
Can you press F12 and check the logs to see if something failed to load?  Looks like it’s missing something…. But not sure what!

**Comment by davepl** [comment]
Sorry, read thread in wrong order :-)

I’ll take a look and let you know if I can think of anything!

**Comment by erikpt** [comment]
Looks like it's there?

(I) (setup)(C1) Setup complete - ESP32 Free Memory: 60760
(I)
(I) (loop)(C1) IP: 192.168.0.109, Mem: 60528 LargestBlk: 41972 PSRAM Free: 0/0 Buffer: 0/21 LED FPS: 28, LED Watt: 30, LED Brite: 100%, CPU: 12%, 02%, FreeDraw: 0.014946
**List All Files
FILE: asset-manifest.json
FILE: index.html
FILE: main.0f983434.css
FILE: main.js.gz
FILE: main.js.LICENSE.txt**
(I) (loop)(C1) IP: 192.168.0.109, Mem: 59820 LargestBlk: 41972 PSRAM Free: 0/0 Buffer: 0/21 LED FPS: 27, LED Watt: 30, LED Brite: 100%, CPU: 15%, 03%, FreeDraw: 0.014946
(I) (loop)(C1) IP: 192.168.0.109, Mem: 59820 LargestBlk: 41972 PSRAM Free: 0/0 Buffer: 0/21 LED FPS: 27, LED Watt: 30, LED Brite: 100%, CPU: 11%, 02%, FreeDraw: 0.014946
**List All Files
FILE: asset-manifest.json
FILE: index.html
FILE: main.0f983434.css
FILE: main.js.gz
FILE: main.js.LICENSE.txt**
(I) (loop)(C1) IP: 192.168.0.109, Mem: 59820 LargestBlk: 41972 PSRAM Free: 0/0 Buffer: 0/21 LED FPS: 28, LED Watt: 30, LED Brite: 100%, CPU: 12%, 02%, FreeDraw: 0.011995

**Reply by erikpt** [reply]
Adding further context.  The webserver will serve up any content as long as it's not in the gzip.

![image](https://user-images.githubusercontent.com/4423348/205188557-0727fad6-5986-4b8d-a955-bc48f5baafc7.png)

![image](https://user-images.githubusercontent.com/4423348/205188625-a5a42fb3-37a7-4ea5-a003-f6168d3fa29a.png)

**Comment by altoidaddict** [comment]
I have this same issue.  I couldn't get the webserver to work at all and then realized perhaps its an issue enabling with ledstrip so I changed over to Spectrum and now just get the yellow screen. I'm on an M5 CPlus.  I was able to get past the dreaded blank yellow screen by changing the audioserial define to 0 from 1.  Crashes completely on ledstrip (similar to issue 73).  However, now I am getting a basic page with a bunch of latin.  Here is a screenshot
![ND_Web](https://user-images.githubusercontent.com/88634078/207136001-5a21e4aa-c318-4510-841b-2e5008b153a7.png)

**Comment by burginh** [comment]
That's the same screen I get, with the latin.
